### ISSUE TITLE
Lualatex engine fails, ignores output directory

### ISSUE THREAD

**Issue by kapep** [jgm/pandoc#1833]
When using the lualatex engine I get this error: `Error producing PDF from TeX source`

I've used following command with a very simple test file. The pdflatex and xelatex engines are working fine.

    >pandoc -f markdown -t latex --latex-engine=lualatex -o foo.pdf test.md

It seems to ignore the temporary output directory and creates the files `input.aux` `input.log` `input.out` and `input.pdf ` in the current directory.

It might be an issue of lualatex. For now I've used this workaround: Create and use `lulatex.cmd` as engine which moves the generated file to the temp directory:

    @echo off
    "C:\Program Files (x86)\MiKTeX 2.9\miktex\bin\lualatex.exe" %*
    move input.aux %5
    move input.log %5
    move input.out %5
    move input.pdf %5

System: pandoc 1.13.2, LuaLaTeX, Version beta-0.76.0-2013062820  (MiKTeX 2.9) (rev 4627), Windows 8.1

**Comment by ekmecic** [comment]
I am getting the same error, and have been getting it for some time on Windows. Upon compilation, the desired document is created (with the name "input.pdf") as well as a LuaLaTeX log, which throws no errors. pandoc returns the error `Error producing PDF from TeX source`.

I have tried to use LuaLaTeX and pandoc together on different Windows systems, but no configurations output successfully. pdfLaTeX and XeLaTeX work properly. Linux works fine with LuaLaTeX as well, so it seems this is isolated to Windows + LuaLaTeX in my case.

I'd be happy to provide a log and additional information if anyone needs it.

System:
pandoc 1.13.2
LuaTeX, Version beta-0.79.1 (MiKTeX 2.9 64-bit) (rev 4971)
Windows 8.1 Pro x64

**Comment by jgm** [comment]
Yes, a log would be great.  I only have a minimal Windows
setup for building pandoc binaries, so it's hard for me to
test effectively.

**Comment by ekmecic** [comment]
Below is the link to the Gist with both the LuaLaTeX log (scrubbed of personal info) and test.md input file. I ran: `pandoc --latex-engine=lualatex test.md -o test.pdf`

https://gist.github.com/ekmecic/9b254c4022538f29e956

**Comment by ekmecic** [comment]
My logs above were on a system with 8+ character username, so I thought it might be related to #777. However I have tried to compile again on a separate Windows system and the same errors mentioned above occur. Specifically, the document is created but ends up in the wrong directory, and pandoc throws an error.
I'd be happy to add any relevant logs or outputs if they would help.

**Comment by nkalvi** [comment]
@ekmecic
I tried specifying the output directory under Windows (using short paths also) with f.ex. 
`d:\>lualatex --output-directory o j.tex` 
and many other possible variants; and as @kapep pointed out, LuaLaTex seems to ignore
`--output-directory` directive and output is left in the current directory.

I'm not sure whether posting this issue in LuaTex mailing lists would be helpful.

----
However, I wonder whether temporarily switching to the temp-dir would be a good solution.
F.ex. this change in PDF.hs seems to eliminate the problem:

```
    cd <- getCurrentDirectory
    setCurrentDirectory tmpDir'
    (exit, out, err) <- pipeProcess (Just env'') program programArgs BL.empty
    setCurrentDirectory cd
```
Not sure how this will affect the other parts/cases though.


BTW, I tested with
LuaTeX, Version beta-0.79.1 (MiKTeX 2.9 64-bit) (rev 4971)
Windows 8.1 Pro x64

**Comment by jgm** [comment]
+++ nkalvi [May 03 15 07:55 ]:
>@ekmecic
>I tried specifying the output directory under Windows (using short paths also) with f.ex.
>`d:\>lualatex --output-directory o j.tex`
>and many other possible variants; and as @kapep pointed out, LuaLaTex seems to ignore
>`--output-directory` directive and output is left in the current directory.
>
>I'm not sure whether posting this issue in LuaTex mailing lists would be helpful.
>
>----
>However, I wonder whether temporarily switching to the temp-dir would be a good solution.
>F.ex. this change in PDF.hs seems to eliminate the problem:
>
>```
>    cd <- getCurrentDirectory
>    setCurrentDirectory tmpDir'
>    (exit, out, err) <- pipeProcess (Just env'') program programArgs BL.empty
>    setCurrentDirectory cd
>```
>Not sure how this will affect the other parts/cases though.

This may cause problems for finding images and other files that the document refers to.  If `--output-directory` just doesn't work as advertised with lualatex on windows, then maybe inquiring on the lualatex list would be the best approach?

**Comment by ekmecic** [comment]
A quick search online nets a few results relating to MiKTeX and --output=directory on LuaTeX. It seems to be a bug with MiKTeX's implementation of LuaTeX. I've attached the links below. Looks like there is a possible solution in the second link, but it may be ConTeXt specific.

https://tex.stackexchange.com/questions/145717/lualatex-and-mylatexformat
https://tex.stackexchange.com/questions/67928/how-to-set-output-directory-in-context-command

To investigate, I switched to TeXLive from MiKTeX and tested again. Under TeXLive 2014, pandoc converts flawlessly using LuaTeX. Windows users will have to use TeXLive for LuaTeX compatibility it seems.

Once again, I'd be happy to provide more logs and information.
